### DISCUSSION TITLE
Chat list -> chat detail navigation

### DISCUSSION THREAD

**Discussion by alelordelo** [Yoddikko/yoddChatGPT#8] [Ideas]
Would be great to have all chats saved into a list. Like on Apple messages.

the SwiftUI core data code snippet I submitted should be able to do that

![2362E6E7-E17F-4DDA-B5C4-69D5507D8530](https://user-images.githubusercontent.com/17708751/226111273-7799e9b6-7a80-40e0-a598-0a7b96d611cc.jpeg)

**Comment by Yoddikko** [comment]
Hey, thanks again for posting the idea.

The list you have in mind are like active chats or only saved old messages? Because if it's an active chats like on the web version then I sadly have to refuse your idea suggestion.

If you mean like this (web version):
![image](https://user-images.githubusercontent.com/92546954/226474075-02d9eba2-f77a-471b-a8f8-f69148918efe.png)

then my answer is no. Mainly because it's not intuitive (design speaking).

Because on the web version every istance remembers the past conversation of **that** chat.

Meanwhile in the app it doesn't work like that. Because of the app is built and how the API works, every chat would remember every topic, making the different lists  not intuitive, because I imagine someone using the app would assume that the bot remembers the interaction on **that** specific list. 

For example if I ask a question to the bot in **list A** and then go go **list B** and ask another question, and then go back again to **list A** and ask some clarification on the last message I sent, he would answer to my last message I sent not on the last message I sent in that specific list. So in this case it would send a message about the last message in the **list B**.

**Reply by yas71** [reply]
The suggestion would be valid if you could have multiple ongoing conversations that use special system prompts. In one conversation you could instruct the model to reply as a therapist, in the other as a poet, etc.

**Comment by alelordelo** [comment]
yup,  multiple ongoing conversations is the goal. do you know if that is possible with current API?

**Reply by yas71** [reply]
only tried it out in the Playground, but I think it should be possible with the role parameter:
https://platform.openai.com/docs/guides/chat/instructing-chat-models

**Comment by Yoddikko** [comment]
No I don't think it is possible with the current API implementation. I'm not sure tho because the framework got updated lately. If it's possible I will add for sure :)

**Reply by yas71** [reply]
Awesome, looking forward to it!
